activation/poet.go

```diff
@@ -61,11 +84,22 @@ func withCustomHttpClient(client *http.Client) PoetClientOpts {
 	}
 }
 
+func WithLogger(logger *zap.Logger) PoetClientOpts {
+	return func(c *HTTPPoetClient) {
+		c.logger = logger
+		c.client.Logger = &retryableHttpLogger{inner: logger}
```

Comment:
I don't think this `Logger` is very useful for us, it only prints 2 possible messages:

- `DEBUG` performing request ...
- `ERROR` when failing to drain the request body in the case of an error (that error is not logged with `Logger`

I think we should set `RequestLogHook` instead and not use the `Logger` for the possible performance impact when logging every request on `DEBUG` which `RequestLogHook` would also do and we can use the structured logger instead of the sugared 🙂

Reply:
>I don't think this Logger is very useful for us, it only prints 2 possible messages
Not exactly, check out this example retried request:
```
DEBUG	poet	performing request	{"method": "GET", "url": "http://127.0.0.1:38837/v1/proofs/0"}
INFO	poet	response received	{"url": "http://127.0.0.1:38837/v1/proofs/0", "status": 404} // <-- from ResponseLogHook
DEBUG	poet	retrying request	{"request": "GET http://127.0.0.1:38837/v1/proofs/0 (status: 404)", "timeout": "107.918576ms", "remaining": 10}
INFO	poet	response received	{"url": "http://127.0.0.1:38837/v1/proofs/0", "status": 200}
```

> I think we should set RequestLogHook instead and not use the Logger for the possible performance impact

IMHO it doesn't make **any** difference. There are literally only a handful of requests to PoET.

Reply:
Can you try with setting the `RequestLogHook` instead of the `Logger`. It should print the same (and possibly more) information but not cause any allocations if logging is disabled (while the `retryableHttpLogger` will).